ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcLogAppender.java

```diff
@@ -394,17 +412,24 @@ public void onNext(InstallSnapshotReplyProto reply) {
     @Override
     public void onError(Throwable t) {
       if (!isAppenderRunning()) {
-        LOG.info("{} is stopped", this);
+        LOG.info("{} is stopped", GrpcLogAppender.this);
         return;
       }
-      LOG.error("{}: Failed installSnapshot: {}", this, t);
+      GrpcUtil.warn(LOG, () -> this + ": Failed InstallSnapshot", t);
+      grpcServerMetrics.onRequestRetry(); // Update try counter
       resetClient(null);
       close();
     }
 
     @Override
     public void onCompleted() {
-      LOG.info("{}: follower responses installSnapshot COMPLETED", this);
+      if (!isNotificationOnly) {
+        LOG.info("{}: follower responded installSnapshot COMPLETED", this);
+      } else {
+        if (LOG.isDebugEnabled()) {
+          LOG.info("{}: follower responded installSnapshot COMPLETED", this);
+        }
+      }
```

Comment:
We can combine the if statements using below condition.
`!isNotificationOnly || LOG.isDebugEnabled()`

Reply:
Done.